### ISSUE TITLE
Feature request: rule to enforce asterisks

### ISSUE THREAD

**Issue by rasenplanscher** [gajus/eslint-plugin-jsdoc#199]
It would be nice to have a rule that enforces every line in a JSdoc comment to start with an asterisk, as in `https://palantir.github.io/tslint/rules/jsdoc-format/`

**Comment by golopot** [comment]
Comparison between `tslint/jsdoc-format` and `eslint-plugin-jsdoc`: 
- [ ] each line contains an asterisk
- [x] asterisks must be aligned - jsdoc/check-alignment
- [x] disallow extra space after asterisks - jsdoc/check-indentation
- [ ] require one space after asterisks
- [ ] multiline comments don’t allow text after /** in the first line (under option)

Both unsupported:
- [ ] bare block comment: disallow asterisks

**Comment by skeggse** [comment]
I submitted a PR for this at #446. Not amazingly happy about the name, though - thoughts?

**Comment by brettz9** [comment]
Other checks I can think of on this general topic might be:

1. Avoiding multiple asterisks (beyond the obligatory first two), whether at the beginning, end, or middle
2. Avoiding single line jsdoc blocks such as `/** @param abc */`, perhaps excepting some tags like [`@lends`](https://jsdoc.app/tags-lends.html) which, um, "lend" themselves to single line expressions because of their terseness.

**Comment by brettz9** [comment]
Regarding reporting jsdoc blocks with multiple asterisks at the beginning, I've just pushed a fix at 28397c7ed54b34a320a9d3961131da2dbf87c534 so that `no-bad-blocks` can also check 3+ asterisks when what appears to be a tag is found therein. That case is being handled separately because multiple asterisks at the beginning are not actually jsdoc (so we use that rule to catch comments which *look like* jsdoc).

I've also filed #728 so as to allow preventing `/*** ... */` in that rule even where the contents appear to have no tag (normally not prevented since it could just be a multiline comment with a lot of asterisks).

**Comment by brettz9** [comment]
After the current PRs, I think the one remaining item here would belong in a rule like `multiline-comments` with options;

1. No line 0 text if multiline
2. Avoid single line entirely, unless optionally possessing specific tags like `@lends`
3. Require single lines only, optionally if under a certain length and/or if tags and/or particular tags are present

**Comment by brettz9** [comment]
> After the current PRs, I think the one remaining item here would belong in a rule like `multiline-comments` with options;
> 
>     1. No line 0 text if multiline
> 
>     2. Avoid single line entirely, unless optionally possessing specific tags like `@lends`
> 
>     3. Require single lines only, optionally if under a certain length and/or if tags and/or particular tags are present

Btw, @golopot , it'd be great to have your feedback in before I prepare the PRs. Thanks...

**Comment by danvk** [comment]
I'm not sure whether tslint enforced this, but another asterisk-related formatting issue I sometimes see is not putting the trailing `*/` on its own line in a multiline block.

For example, rather than writing it like this:

```ts
/**
 * This is a multline
 * jsdoc comment */
```

the more canonical way to write it is like this:

```ts
/**
 * This is a multline
 * jsdoc comment
 */
```

It looks like `no-multi-asterisks` is in the ballpark, but doesn't quite enforce this.

**Comment by brettz9** [comment]
Please file as a new issue. Thanks!

**Comment by brettz9** [comment]
And, btw, I think it would be fine as a new option under `no-multiline-blocks` given the rule's already handling single line blocks).

**Comment by danvk** [comment]
@brettz9 sure, filed #738
